### PR TITLE
[ANCHOR-636] Clean up deprecation for `fee_details` 

### PR DESCRIPTION
**Pull request by JiahuiWho**
### Description

Clean up the `amount_fee` and `amount_fee` in SEP-6/24/31 Transaction object and appropriate RPC endpoints

### Testing

- `./gradlew test`

### CODE REVIEW

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep31/Sep31GetTransactionResponse.java [hunk 1]
```diff
@@ -44,11 +45,8 @@ public static class TransactionResponse {
     @SerializedName("amount_out_asset")
     String amountOutAsset;
 
-    @SerializedName("amount_fee")
-    String amountFee;
-
-    @SerializedName("amount_fee_asset")
-    String amountFeeAsset;
+    @SerializedName("fee_details")
+    FeeDetails feeDetails;
```
Comment (Ifropc) [inline api-schema/src/main/java/org/stellar/anchor/api/sep/sep31/Sep31GetTransactionResponse.java:49]: Was it missing in the response? Can you port addition of this field back to v2? And can you also please add a test to check that feeDetails is present in SEP6/24/31 responses? (I think I added them but I might've missed some)

File: platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyOffchainFundsReceivedHandler.java [hunk 2]
```diff
@@ -74,28 +74,21 @@ protected void validate(JdbcSepTransaction txn, NotifyOffchainFundsReceivedReque
     // None of the amounts are provided
     ((request.getAmountIn() == null
             && request.getAmountOut() == null
-            && request.getAmountFee() == null
             && request.getFeeDetails() == null)
         ||
         // All the amounts are provided (allow either amount_fee or fee_details)
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyOffchainFundsReceivedHandler.java:79]: Can you update this comment please?

File: platform/src/test/kotlin/org/stellar/anchor/platform/service/TransactionServiceTest.kt [hunk 3]
```diff
@@ -1123,25 +1122,26 @@ class TransactionServiceTest {
...
-
-    data.feeDetails = Amount("9", "USDC").toRate()
-    var ex = assertThrows<BadRequestException> { transactionService.validateAndGetRateFee(data) }
-    assertEquals("amount_fee's amount doesn't match amount from fee_details", ex.message)
-
-    data.feeDetails = Amount("10", "NOTUSDC").toRate()
-    ex = assertThrows<BadRequestException> { transactionService.validateAndGetRateFee(data) }
-    assertEquals("amount_fee's asset doesn't match asset from fee_details", ex.message)
-  }
+  //  @Test
```
Comment (Ifropc) [inline platform/src/test/kotlin/org/stellar/anchor/platform/service/TransactionServiceTest.kt:1125]: Should it be un-commented?

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep31/Sep31GetTransactionResponse.java (hunk 1 above)
Comment (JiahuiWho) [reply api-schema/src/main/java/org/stellar/anchor/api/sep/sep31/Sep31GetTransactionResponse.java:49]: Yeah it was missing in SEP31 response. Will double check on the tests.

File: platform/src/main/java/org/stellar/anchor/platform/rpc/RequestOnchainFundsHandler.java [hunk 4]
```diff
@@ -143,9 +131,7 @@ protected void validate(JdbcSepTransaction txn, RequestOnchainFundsRequest reque
     if (request.getAmountOut() == null && txn.getAmountOut() == null) {
       throw new InvalidParamsException("amount_out is required");
     }
-    if (request.getAmountFee() == null
-        && request.getFeeDetails() == null
-        && txn.getAmountFee() == null) {
+    if (request.getFeeDetails() == null && txn.getAmountFee() == null) {
```
[the remark this answers is not in the document]
Comment (JiahuiWho) [reply platform/src/main/java/org/stellar/anchor/platform/rpc/RequestOnchainFundsHandler.java:134]: This `txn` is `JdbcSepTransaction`, where we store `amountFee`, `amountFeeAsset`and `feeDetailsList` (the 3 fields of `feeDetails`) separatly

File: extended-tests/src/test/kotlin/org/stellar/anchor/platform/extendedtest/custody/CustodyApiTests.kt [hunk 5]
```diff
@@ -662,15 +662,15 @@ private const val NOTIFY_OFFCHAIN_FUNDS_RECEIVED_REQUEST =
     "message": "test message",
     "amount_in": {
         "amount": "1",
-        "asset": "stellar:USDC:GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5"
+        "asset": "iso4217:USD"
```
[the remark this answers is not in the document]
Comment (JiahuiWho) [reply extended-tests/src/test/kotlin/org/stellar/anchor/platform/extendedtest/custody/CustodyApiTests.kt:665]: No this is just to match `REQUEST_OFFCHAIN_FUNDS_REQUEST`

File: platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyAmountsUpdatedHandler.java [hunk 6]
```diff
@@ -68,19 +62,11 @@ protected void validate(JdbcSepTransaction txn, NotifyAmountsUpdatedRequest requ
...
-              .amount(request.getAmountFee().getAmount())
-              .asset(txn.getAmountFeeAsset())
-              .build(),
-          true,
-          assetService);
-    }
-    if (request.getFeeDetails() != null) {
-      AssetValidationUtils.validateFeeDetails(request.getFeeDetails(), txn, assetService);
+
+    if (request.getFeeDetails() == null) {
```
[the remark this answers is not in the document]
Comment (JiahuiWho) [reply platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyAmountsUpdatedHandler.java:66]: `fee_details` has been updated to doc, will remove `amountFee` and `amountFeeAsset` once 3.0 launched
Tracking in https://stellarorg.atlassian.net/browse/ANCHOR-677

File: core/src/main/java/org/stellar/anchor/util/TransactionHelper.java [hunk 7]
```diff
@@ -131,22 +129,9 @@ public static GetTransactionResponse toGetTransactionResponse(
...
-                : null)
-        .amountOut(
-            (amountOutAsset != null && txn.getAmountOut() != null)
-                ? Amount.create(txn.getAmountOut(), amountOutAsset)
-                : null)
-        .amountFee(
-            (amountFeeAsset != null && txn.getAmountFee() != null)
-                ? Amount.create(txn.getAmountFee(), amountFeeAsset)
-                : null)
+        .amountExpected(new Amount(txn.getAmountExpected(), amountExpectedAsset))
```
[the remark this answers is not in the document]
Comment (JiahuiWho) [reply core/src/main/java/org/stellar/anchor/util/TransactionHelper.java:132]: I will revert this change, same as below. 
These was introduced in the merge commit, probably the result of resolving some conflicts

File: extended-tests/src/test/kotlin/org/stellar/anchor/platform/extendedtest/custody/CustodyApiTests.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (philipliu) [reply extended-tests/src/test/kotlin/org/stellar/anchor/platform/extendedtest/custody/CustodyApiTests.kt:665]: But I don't understand why the `amount_in` asset would change from removing `amount_fee`.

[the remark this answers is not in the document]
Comment (JiahuiWho) [reply extended-tests/src/test/kotlin/org/stellar/anchor/platform/extendedtest/custody/CustodyApiTests.kt:665]: Only fee_details is realted to the test failure. But I fiigure out NOTIFY_OFFCHAIN_FUNDS_RECEIVED_REQUEST should match REQUEST_OFFCHAIN_FUNDS_REQUEST, so changed both of them.

**Review by philipliu — APPROVED** [review]
LGTM

**Pull request merged**